src/include/storage/data_table.h

```diff
@@ -207,6 +207,12 @@ class DataTable {
    */
   DataTableCounter *GetDataTableCounter() { return &data_table_counter_; }
 
+  /**
+   * Returns a read-only view of this DataTable's BlockLayout.
+   * @return this DataTable's BlockLayout.
+   */
+  const BlockLayout &GetBlockLayout() const { return accessor_.GetBlockLayout(); }
```

Comment:
Similar reason as https://github.com/cmu-db/terrier/pull/341#discussion_r282733370. Currently, when transactions want to log, they only provide DeltaRecords and the DataTable. And most transactions have no idea about SqlTable. (In fact, except the tests written by us, almost all tests directly work on DataTables. They even do not create a SqlTable object.) Therefore, we currently are unable to avoid the DataTable level things.